pkg/oc/cli/admin/release/extract.go

```diff
@@ -249,3 +274,169 @@ func (o *ExtractOptions) extractGit(dir string) error {
 	}
 	return nil
 }
+
+func (o *ExtractOptions) extractCommand(command string) error {
+	currentOS := runtime.GOOS
+	if len(o.CommandOperatingSystem) > 0 {
+		currentOS = o.CommandOperatingSystem
+	}
+	var image string
+	var imagePath string
+	switch command {
+	case "oc":
+		switch currentOS {
+		case "linux":
+			imagePath = "usr/bin/oc"
+			image = "cli"
+		case "darwin", "mac":
+			imagePath = "usr/share/openshift/mac/oc"
+			image = "cli-artifacts"
+		case "windows":
+			imagePath = "usr/share/openshift/windows/oc.exe"
+			image = "cli-artifacts"
+		default:
+			return fmt.Errorf("command %q does not support the operating system %q", o.Command, currentOS)
+		}
+	case "openshift-install":
+		switch currentOS {
+		case "linux":
+			imagePath = "usr/bin/openshift-install"
+			image = "installer"
+		case "darwin", "mac":
+			imagePath = "usr/share/openshift/mac/openshift-install"
+			image = "installer-artifacts"
+		default:
+			return fmt.Errorf("command %q does not support the operating system %q", o.Command, currentOS)
+		}
+	default:
+		return fmt.Errorf("the supported commands are 'oc' and 'openshift-install'")
+	}
+
+	infoOptions := NewInfoOptions(o.IOStreams)
+	release, err := infoOptions.LoadReleaseInfo(o.From, false)
+	if err != nil {
+		return err
+	}
+	spec, err := findImageSpec(release.References, image, o.From)
+	if err != nil {
+		return err
+	}
+	glog.V(2).Infof("Will extract %s from %s", imagePath, spec)
+	ref, err := imagereference.Parse(spec)
+	if err != nil {
+		return err
+	}
+
+	dir := o.Directory
+	if err := os.MkdirAll(dir, 0755); err != nil {
```

Comment:
I'd rather use 0777 here and rely on the user's [umask][1] to limit permissions (maybe they want to share write access with their group, or maybe they don't even want to share read access with their group).

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/umask.html

Reply:
We should audit other MkdirAlls, I don't think this is the only place we do that.

Reply:
I made mkdirall consistent within release commands